extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php

```
@@ -9,161 +9,144 @@ class EmbeddableDiscussionsController {
-		\Wikia::addAssetsToOutput( 'embeddable_discussions_js' );
-		\Wikia::addAssetsToOutput( 'embeddable_discussions_scss' );
+		$modelData = ( new DiscussionsThreadModel( $wgCityId ) )
+			->getData( $parameters[ static::PARAM_MOSTRECENT ], $parameters[ static::PARAM_SIZE ], $parameters[ static::PARAM_CATEGORY ] );
 
-		JSMessages::enqueuePackage( 'EmbeddableDiscussions', JSMessages::EXTERNAL );
+		$errorMessage = $this->validateCategory( $args, $modelData ) ?? $errorMessage;
 
-		return true;
+		if ( isset( $errorMessage ) ) {
+			return $this->templateEngine
+				->clearData()
+				->setData( [ 'errorMessage' => $errorMessage ] )
+				->render( 'DiscussionError.mustache' );
+		}
+
+		return F::app()->checkSkin( 'wikiamobile' ) ?
+			$this->renderMobile( $modelData, $parameters[ static::PARAM_MOSTRECENT ], $parameters[ static::PARAM_SIZE ] ) :
+			$this->renderDesktop( $modelData, $parameters[ static::PARAM_MOSTRECENT ], $parameters[ static::PARAM_CATEGORY ], $parameters[ static::PARAM_COLUMNS ] );
 	}
 
 	/**
-	 * Checks arguments for errors.
-	 * @param array $args
-	 * @param string errorMessage Return parameter with the proper error message to show. Disregard if return is false
-	 * @return true if ok, false if error
+	 * @param array $parameters
+	 * @return String error message or empty string if category is valid
 	 */
-	private function checkArguments( array $args, $modelData, &$errorMessage ) {
-		// mostrecent must be bool
-		if ( isset( $args['mostrecent'] ) &&
-			$args['mostrecent'] !== 'true' &&
-			$args['mostrecent'] !== 'false'
-		) {
-			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'mostrecent' )->plain() .
-				wfMessage( 'embeddable-discussions-parameter-error-boolean' )->plain();
-
-			return false;
+	private function validateArguments( array $parameters ) {
+		$errorMessage = '';
+
+		// PARAM_MOSTRECENT must be bool
+		if ( !AttributesValidator::isBoolish( $parameters[ static::PARAM_MOSTRECENT ] ) ) {
```

We could return error if this fails instead of continuing.